### ISSUE TITLE
Crashes in Vite watch mode with consecutive quick changes

### ISSUE THREAD

**Issue by segevfiner** [qmhc/vite-plugin-dts#193]
### Describe the bug

In a large project where building the d.ts takes long enough, making another change while the plugin is still building leads to an error of this kind:
```
[Error: ENOENT: no such file or directory, mkdir 'project/dist/components'] {
  errno: -2,
  code: 'PLUGIN_ERROR',
  syscall: 'mkdir',
  path: 'project/dist/components',
  pluginCode: 'ENOENT',
  plugin: 'vite:dts',
  hook: 'closeBundle'
}
```

### Reproduction

https://stackblitz.com/edit/vitejs-vite-mt6xsz?file=README.md

### Steps to reproduce

1. Open another terminal.
2. Run `touch src/components/HelloWorld.vue` in a loop. Basically up, enter, until the error occurs. (It happens from normal editing in larger projects where vite-plugin-dts takes longer to run)

### System Info

```shell
System:
    OS: macOS 13.2.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 22.59 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.19.1 - ~/.nvm/versions/node/v16.19.1/bin/node
    npm: 9.6.2 - ~/.nvm/versions/node/v16.19.1/bin/npm
  Browsers:
    Chrome: 111.0.5563.64
    Safari: 16.3
  npmPackages:
    @vitejs/plugin-vue: ^4.0.0 => 4.0.0 
    vite: ^4.0.0 => 4.1.4 
    vite-plugin-dts: 2.1.0 => 2.1.0
```

### Validations

- [X] Read the [FAQ](https://github.com/qmhc/vite-plugin-dts#faq).
- [X] Check that there isn't [already an issue](https://github.com/qmhc/vite-plugin-dts/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by lobaak** [comment]
I'm also seeing this issue with a large library

**Comment by StringKe** [comment]
The same problem is encountered with watch mode when an exception is thrown after a file is changed.

**Comment by zbouslikhin** [comment]
Still happening as per the 12th of June 2023

**Comment by qmhc** [comment]
Fixed in `3.0.0`.
